api/src/setup/index.ts

```diff
@@ -1,113 +1,204 @@
-		logger.log('==================================================================================================');
+        if (!this.serverIdentityId) {
+            throw Error('You need to specify a server identity file (SERVER_IDENTITY)');
+        }
+    }
+        
+    // Ensure that on the db there is the declared root identity
+    async checkRootIdentity() : Promise<IdentityJsonUpdate> {
+    
+        logger.log(`Checking root identity...`);
+    
+        const serverIdentityId = this.keyResolver.resolve(this.serverIdentityId);
+        if (!serverIdentityId) {
+            logger.error("Root identity is missing")
+            return null;
+        }
+    
+        const ssiService = SsiService.getInstance(this.identityConfig, logger);
+        const userService = new UserService(ssiService, this.serverSecret, logger);
 
-		// re-create the verification service with a valid server identity id
-		const verificationService = new VerificationService(
-			ssiService,
-			userService,
-			{
-				serverSecret,
-				serverIdentityId: identity.doc.id,
-				keyCollectionSize: KEY_COLLECTION_SIZE
-			},
-			logger
-		);
+        const tmpVerificationService = new VerificationService(
+            ssiService,
+            userService,
+            {
+                serverSecret: this.serverSecret,
+                serverIdentityId,
+                keyCollectionSize: KEY_COLLECTION_SIZE
+            },
+            logger
+        );
+    
+        const serverIdentity = await tmpVerificationService.getIdentityFromDb(serverIdentityId);
+    
+        if (!serverIdentity) {
+            throw Error('Root identity not found in database: ' + serverIdentityId);
+        }
+        
+        return serverIdentity;
+    
+    }
+    
+    // Check if identity is a valid one
+    private async verifyIdentity(serverIdentity: IdentityJsonUpdate) {
+    
+        // verify if secret key of the server can be used to sign and verify a challenge
+        // if the secret key was changed the server won't be able to decrypt the secret key of the server
+        // and thus is not able to verify the challenge
+        logger.log('Check if server has valid keypair...');
+        const nonce = createNonce();
+        let verified = false;
+        try {
+            const signedNonce = await signNonce(getHexEncodedKey(serverIdentity.key.secret), nonce);
+            verified = await verifySignedNonce(getHexEncodedKey(serverIdentity.key.public), nonce, signedNonce);
+        } catch (e) {
+            logger.error('error when signing or verifying the nonce, the secret key might have changed...');
+        }
+        if (!verified) {
+            throw Error('server keys cannot be verified!');
+        }
+    
+        logger.log('Api is ready to use!');
+    
+    }
+    
+    private async getRootIdentityFromId(serverIdentityId: string) : Promise<IdentityJsonUpdate> {
+    
+        // TODO create database, documents and indexes in mongodb at the first time!
```

Comment:
Is this still a todo? if so please at the ticket number to the todo otherwise remove it please

Reply:
I added a comment in #254. I just renamed the comment